lib/streamlit/runtime/app_session.py

```diff
@@ -349,6 +349,15 @@ def request_rerun(self, client_state: ClientState | None) -> None:
             to use previous client state.
 
         """
+
+        if (
```

Comment:
I think we also need to fill in the RerunData in below with the context_info. Otherwise, it won't work if you do backend-based rerun via st.rerun

Reply:
Done